### PR TITLE
Deprecate top level access to legacy device base classes

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

`qml.Device`, `qml.QubitDevice`, and `qml.QutritDevice` all reflect the legacy device interface, which is no longer the recommended way of creating devices.  

**Description of the Change:**

Deprecate top level access to the `Device`, `QubitDevice`, and `QutritDevice`.

**Benefits:**

Further isolation of the legacy device interface.

**Possible Drawbacks:**

All deprecations propagate through the ecosystem and can cause issues.

**Related GitHub Issues:**

[sc-71519]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -83,6 +83,10 @@
 
 <h3>Deprecations 👋</h3>
 
+* `Device`, `QubitDevice`, and `QutritDevice` will no longer be accessible via top-level import in v0.40.
+  They will still be accessible as `qml.devices.LegacyDevice`, `qml.devices.QubitDevice`, and `qml.devices.QutritDevice`
+  respectively.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:88]:
```suggestion
  respectively.
  [(#6238)](https://github.com/PennyLaneAI/pennylane/pull/6238/)
```

File: tests/test_vqe.py [hunk 2]
```diff
@@ -200,22 +200,30 @@ def amp_embed_and_strong_ent_layer(params, wires=None):
...
+        )
+        m.setattr(qml.devices.LegacyDevice, "short_name", "MockDevice")
+        m.setattr(qml.devices.LegacyDevice, "expval", lambda self, x, y, z: 1)
+        m.setattr(qml.devices.LegacyDevice, "var", lambda self, x, y, z: 2)
+        m.setattr(qml.devices.LegacyDevice, "sample", lambda self, x, y, z: 3)
+        m.setattr(qml.devices.LegacyDevice, "apply", lambda self, x, y, z: None)
 
         def get_device(wires=1):
-            return qml.Device(wires=wires)  # pylint:disable=abstract-class-instantiated
+            return qml.LegacyDevice(wires=wires)  # pylint:disable=abstract-class-instantiated
```
Comment (astralcai) [inline tests/test_vqe.py:226]:
```suggestion
            return qml.devices.LegacyDevice(wires=wires)  # pylint:disable=abstract-class-instantiated
```

Comment (astralcai) [reply tests/test_vqe.py:226]: I'm curious why no deprecation warnings are raised from here.

Comment (albi3ro) [reply tests/test_vqe.py:226]: Looks like it isn't getting used by anything... I'm just deleting the fixture.

**Review by astralcai — APPROVED** [review]

File: .github/workflows/interface-unit-tests.yml [hunk 3]
```diff
@@ -379,8 +379,7 @@ jobs:
       # catalyst requires the latest version of pennylane that is about to be released.
       # Installing catalyst after pennylane to make sure that the latest catalyst is used.
       install_catalyst_nightly: true
-      # using lightning master does not work for the tests with external libraries
-      install_pennylane_lightning_master: false
+      install_pennylane_lightning_master: true
```
Comment (mudit2812) [inline .github/workflows/interface-unit-tests.yml:382]: whoops 😬

File: doc/development/deprecations.rst [hunk 4]
```diff
@@ -9,6 +9,13 @@ deprecations are listed below.
 Pending deprecations
 --------------------
 
+* `Device`, `QubitDevice`, and `QutritDevice` will no longer be imported top level in v0.40.  They instead
+  we be available as `qml.devices.LegacyDevice`, `qml.devices.QubitDevice`, and `qml.devices.QutritDevice`
```
Comment (mudit2812) [inline doc/development/deprecations.rst:13]:
```suggestion
* ``Device``, ``QubitDevice``, and ``QutritDevice`` will no longer be imported top level in v0.40.  They instead
  we be available as ``qml.devices.LegacyDevice``, ``qml.devices.QubitDevice``, and ``qml.devices.QutritDevice``
```

File: pennylane/__init__.py [hunk 5]
```diff
@@ -180,13 +178,30 @@ def __getattr__(name):
     if name == "plugin_devices":
         return pennylane.devices.device_constructor.plugin_devices
 
+    import warnings  # pylint: disable=redefined-outer-name, import-outside-toplevel
```
Comment (mudit2812) [inline pennylane/__init__.py:181]: Could we just do the following to get rid of the first pylint error? And then update the warnings below to match

```suggestion
    from warnings import warn  # pylint: disable=import-outside-toplevel
```

**Review by mudit2812 — APPROVED** [review]

File: pennylane/__init__.py (hunk 5 above)
Comment (albi3ro) [reply pennylane/__init__.py]: changed locally.

**Pull request merged**